lib/phoenix_live_view/html_tokenizer.ex

```diff
@@ -59,156 +59,212 @@ defmodule Phoenix.LiveView.HTMLTokenizer do
-    handle_script(rest, line, column + 1, [char_or_bin(c) | buffer], acc, state)
+  defp handle_script(<<c::utf8, rest::binary>>, line_offset, column, buffer, acc, state) do
+    handle_script(rest, line_offset, column + 1, [char_or_bin(c) | buffer], acc, state)
   end
 
-  defp handle_script(<<>>, line, column, buffer, acc, _state) do
-    ok(text_to_acc(buffer, acc, line, column, []), :script)
+  defp handle_script(<<>>, line_offset, column, buffer, acc, _state) do
+    ok(text_to_acc(buffer, acc, line_offset, column, []), :script)
   end
 
   ## handle_style
 
-  defp handle_style("</style>" <> rest, line, column, buffer, acc, state) do
+  defp handle_style("</style>" <> rest, {line, offset}, column, buffer, acc, state) do
     acc = [
       {:tag_close, "style", %{line: line, column: column}}
-      | text_to_acc(buffer, acc, line, column, [])
+      | text_to_acc(buffer, acc, {line, offset}, column, [])
     ]
 
-    handle_text(rest, line, column + 9, [], acc, state)
+    handle_text(rest, {line, offset}, column + 9, [], acc, state)
   end
 
-  defp handle_style("\r\n" <> rest, line, _column, buffer, acc, state) do
-    handle_style(rest, line + 1, state.column_offset, ["\r\n" | buffer], acc, state)
+  defp handle_style("\r\n" <> rest, {line, offset}, column, buffer, acc, state) do
+    handle_style(
+      rest,
+      {line + 1, offset + column - state.column_offset + 2},
+      state.column_offset,
+      ["\r\n" | buffer],
+      acc,
+      state
+    )
   end
 
-  defp handle_style("\n" <> rest, line, _column, buffer, acc, state) do
-    handle_style(rest, line + 1, state.column_offset, ["\n" | buffer], acc, state)
+  defp handle_style("\n" <> rest, {line, offset}, column, buffer, acc, state) do
+    handle_style(
+      rest,
+      {line + 1, offset + column - state.column_offset + 1},
+      state.column_offset,
+      ["\n" | buffer],
+      acc,
+      state
+    )
   end
 
-  defp handle_style(<<c::utf8, rest::binary>>, line, column, buffer, acc, state) do
-    handle_style(rest, line, column + 1, [char_or_bin(c) | buffer], acc, state)
+  defp handle_style(<<c::utf8, rest::binary>>, line_offset, column, buffer, acc, state) do
+    handle_style(rest, line_offset, column + 1, [char_or_bin(c) | buffer], acc, state)
   end
 
-  defp handle_style(<<>>, line, column, buffer, acc, _state) do
-    ok(text_to_acc(buffer, acc, line, column, []), :style)
+  defp handle_style(<<>>, line_offset, column, buffer, acc, _state) do
+    ok(text_to_acc(buffer, acc, line_offset, column, []), :style)
   end
 
   ## handle_comment
 
-  defp handle_comment("\r\n" <> rest, line, _column, buffer, acc, state) do
-    handle_comment(rest, line + 1, state.column_offset, ["\r\n" | buffer], acc, state)
+  defp handle_comment("\r\n" <> rest, {line, offset}, _column, buffer, acc, state) do
+    handle_comment(rest, {line + 1, offset}, state.column_offset, ["\r\n" | buffer], acc, state)
```

Comment:
You need to do the same dance for newlines here as in other places. I suggest creating a `new_line` helper you can use everywhere like this:

    new_line(line_offset, column, extra, state)

where:

    defp new_line({line, offset}, column, extra, state) do
      {line + 1, offset + column - state.column_offset + extra}
    end